nighthawk/source/client/client.cc

```diff
@@ -187,18 +185,45 @@ class ProcessContext {
     return workers_;
   }
 
-  bool runWorkers() {
+  std::vector<StatisticPtr> vectorizeStatisticPtrMap(const StatisticFactory& statistic_factory,
+                                                     const StatisticPtrMap& statistics) const {
+    std::vector<StatisticPtr> v;
+    for (auto statistic : statistics) {
+      auto new_statistic = statistic_factory.create()->combine(*(statistic.second));
+      new_statistic->setId(statistic.first);
+      v.push_back(std::move(new_statistic));
+    }
+    return v;
+  }
+
+  bool runWorkers(OutputFormatter& formatter) {
+    bool ok = true;
     Envoy::Runtime::RandomGeneratorImpl generator;
     Envoy::Runtime::ScopedLoaderSingleton loader(
         Envoy::Runtime::LoaderPtr{new Envoy::Runtime::LoaderImpl(generator, store(), tls())});
+
     for (auto& w : workers_) {
       w->start();
     }
-    bool ok = true;
     for (auto& w : workers_) {
       w->waitForCompletion();
       ok = ok && w->success();
     }
+
+    if (workers_.size() > 1) {
```

Comment:
Nit: `if (!workers_.empty()) {`

Reply:
Is this even needed, won't the `for` loop nop in the empty case?

Reply:
This is comment-worthy, added.

Reply:
Comment added